src/classes/OPP_OpportunityAmountSync_SVC.cls

```diff
@@ -0,0 +1,32 @@
+public class OPP_OpportunityAmountSync_SVC {
+
+    /**
+     * Put the fields needed in the query in the description.
+     * @param opportunitiesByIds
+     * @return
+     */
+    public List<Opportunity> syncOppAmountWithTotalPaidPaymentAmount(Map<Id, Opportunity> opportunitiesByIds) {
+        List<Opportunity> oppsToUpdate = new List<Opportunity>();
+        
+        for (Opportunity opp : opportunitiesByIds.values()) {
+            Decimal totalPaid = 0;
+
+            for (npe01__OppPayment__c pmt : opp.npe01__OppPayment__r) {
+                if (pmt.npe01__Paid__c) {
```

Comment:
why not make this

totalPaid += (pmt.npe01__Paid__c && pmt.npe01__Payment_Amount__c != null) ? pmt.npe01__Payment_Amount__c : 0;

This eliminates the if, and puts are affirmative choice first.